pkg/generators/servers.go

```diff
@@ -330,19 +331,59 @@ func (g *ServersGenerator) generateServerAdapterSource(resource *concepts.Resour
 			{{ $requestBodyParameters := requestBodyParameters . }}
 			{{ $requestBodyLen := len $requestBodyParameters }}
 			{{ $responseParameters := responseParameters . }}
+			{{ $requestQueryParameters := requestQueryParameters . }}
 	
 			func (a *{{ $adapterName }}) read{{ $requestName }}(r *http.Request) (*{{ $requestName }}, error) {
+				var err error
 				result := new({{ $requestName }})
-				result.query = r.Form
+				result.query = r.URL.Query()
 				result.path = r.URL.Path
 
+				{{ if $requestQueryParameters }}
+					{{ range  $requestQueryParameters }}
+						{{ $fieldName := fieldName . }}
+						{{ $fieldType := fieldType . }}
+						{{ $fieldName }}String := result.query.Get("{{ $fieldName }}")
+						if {{ $fieldName }}String  != "" {
+							{{ if eq .Type.Name.String "integer"}}
+								{{ $fieldName }}, err := strconv.Atoi({{ $fieldName }}String)
```

Comment:
Atoi returns int, whose size depends on platform.
Better use fixed size ParseInt -> int64.
(Probably orthogonal to this issue.  If current CS code is written for api types using int, it'll require some rewrites for int64; need to consider how to do that piece-wise... Is it easier to first covert CS to use generated code, then switch int -> int64?)

Reply:
Will do, thanks!

Reply:
Not sure I follow. Current api uses `int` over `int64`. (both in `ocm-api-model` as well as in `clusters-service`). The way I see it I have two options:

-  Use `int64` here and then cast (truncate) to `int`.
-  Change the api to use `int64` instead.

Reply:
I guess its easier to migrate clusters-service and then change the model and generate a new server.

Reply:
Yeah, this is _way_ out of scope for this PR.  I'll open a separate issue to track.